test/tests/components/content-service/content-service_test.go

```diff
@@ -0,0 +1,317 @@
+// Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package contentservice
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+
+	content_service_api "github.com/gitpod-io/gitpod/content-service/api"
+	"github.com/gitpod-io/gitpod/test/pkg/integration"
+	test_context "github.com/gitpod-io/gitpod/test/pkg/integration/context"
+)
+
+var (
+	gitpodBuiltinUserID = "00000000-0000-0000-0000-000000000000"
+)
+
+func hasErrorCode(err error, code codes.Code) bool {
+	st, ok := status.FromError(err)
+	return ok && st.Code() == code
+}
+
+func TestUploadUrl(t *testing.T) {
+	tests := []struct {
+		Name              string
+		InputOwnerID      string
+		InputName         string
+		ExpectedErrorCode codes.Code
+	}{
+		{
+			Name:         "simple name",
+			InputOwnerID: gitpodBuiltinUserID,
+			InputName:    "test-blob",
+		},
+		{
+			Name:         "new user",
+			InputOwnerID: "new-user",
+			InputName:    "test-blob",
+		},
+		{
+			Name:              "name with whitespace",
+			InputOwnerID:      gitpodBuiltinUserID,
+			InputName:         "whitespaces are not allowed",
+			ExpectedErrorCode: codes.InvalidArgument,
+		},
+		{
+			Name:              "name with invalid char",
+			InputOwnerID:      gitpodBuiltinUserID,
+			InputName:         "ä-is-not-allowed",
+			ExpectedErrorCode: codes.InvalidArgument,
+		},
+	}
+
+	uploadUrlRequest := features.New("UploadUrlRequest").
```

Comment:
There is some variability in the feature naming throughout this PR and given they can be used to filter tests we should standardise on feature naming, i.e. camelcase (or not), spaces Y/N, should they always start with the package to allow grouping, etc.

Reply:
Yes, that's one the missing definitions